### PR TITLE
Updated python version requested from the ci

### PR DESCRIPTION
**Pull request by Iximiel**
##### Description

Python 3.7 has finally come to its EOL,I'm starting a series of PR for trying to see if python 3.9 can be used without to many modifications

##### Copyright

- [x] I agree to transfer the copyright of the code I have written to the PLUMED developers or to the author of the code I am modifying.

- [ ] the module I added or modified contains a `COPYRIGHT` file with the correct license information. Code should be released under an open source license. I also used the command `cd src && ./header.sh mymodulename` in order to make sure the headers of the module are correct. 

##### Tests

- [ ] I added a new regtest or modified an existing regtest to validate my changes.
- [ ] I verified that all regtests are passed successfully on [GitHub Actions](https://github.com/plumed/plumed2/actions).

### CODE REVIEW

**Comment by GiovanniBussi** [comment]
Great!

Looks like with centos 7 this is [not trivial](https://discuss.python.org/t/is-there-any-alternative-way-to-install-python-3-9-in-centos-7-instead-of-using-source-code-tar-gz-tgz/22938/10).

However, I checked a [previously running job](https://github.com/plumed/plumed2/actions/runs/5566584934/job/15075142615) and it looks like the breaking change was the switch to cython 3.0.

Maybe it's sufficient to specify `cython<3` [here](https://github.com/plumed/plumed2/blob/0493ec1266ea10c365229c09313002f0042191c9/docker/centos7#L13)?

**Comment by Iximiel** [comment]
> Maybe it's sufficient to specify `cython<3` [here](https://github.com/plumed/plumed2/blob/0493ec1266ea10c365229c09313002f0042191c9/docker/centos7#L13)?

If it works I think it can be safely merged into 2.9 and then we should carefully update the master, because of the change in the CI

File: docker/centos7
Comment (GiovanniBussi) [inline docker/centos7:1]: I think you need quotes (as for `gsd` in the following line)

Comment (Iximiel) [reply docker/centos7:1]: I saw and I added them

**Pull request merged**